# Data Description
#### Broad Question Prompt from assignment: 
What player characteristics and behaviours are predictive of subscribing to the game-related newsletter?

#### Specific Question I formulated: 
Can player demographics and play behaviour (ex: total playtime, number of sessions, player type) predict whether a player subscribes to the newsletter?

#### Dataset to be used: players.csv
since sessions.csv only has data relevant to playtime analyses, while I want to figure out how player newsletter subscription relates to variables like experience, time, and age in each player

## 0.0) Loading in the data

In [1]:
getwd()

[1] "/home/jovyan/work/swaggest-pino"

In [2]:
list.files()

[1] "Figuring it out.ipynb"       "Individual Planning Project"
[3] "loadingstuff.ipynb"          "players.csv"                
[5] "README.md"                   "sessions.csv"               
[7] "Untitled.ipynb"

In [3]:
library(tidyverse)
sessions_original <- read_csv("sessions.csv")
players_original <- read_csv("players.csv")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter:

In [4]:
print(players_original, rows = 10)

# A tibble: 196 × 7
   experience subscribe hashedEmail              played_hours name  gender   Age
   <chr>      <lgl>     <chr>                           <dbl> <chr> <chr>  <dbl>
 1 Pro        TRUE      f6daba428a5e19a3d475748…         30.3 Morg… Male       9
 2 Veteran    TRUE      f3c813577c458ba0dfef809…          3.8 Chri… Male      17
 3 Veteran    FALSE     b674dd7ee0d24096d1c0196…          0   Blake Male      17
 4 Amateur    TRUE      23fe711e0e3b77f1da7aa22…          0.7 Flora Female    21
 5 Regular    TRUE      7dc01f10bf20671ecfccdac…          0.1 Kylie Male      21
 6 Amateur    TRUE      f58aad5996a435f16b0284a…          0   Adri… Female    17
 7 Regular    TRUE      8e594b8953193b26f498db9…          0   Luna  Female    19
 8 Amateur    FALSE     1d2371d8a35c8831034b25b…          0   Emer… Male      21
 9 Amateur    TRUE      8b71f4d66a38389b7528bb3…          0.1 Nata… Male      47
10 Veteran    TRUE      bbe2d83de678f519c4b3daa…          0   Nyla  Female    22
# ℹ 186 

In [5]:
players_original <- tibble(players_original)

In [33]:
#(this is just the selection of variables i'll be using later, just thought it was more tidy to store it up here)
player_data <- players_original |>
    select(experience, subscribe, played_hours, gender, Age)
player_data

experience subscribe played_hours gender            Age
1   Pro         TRUE     30.3         Male               9 
2   Veteran     TRUE      3.8         Male              17 
3   Veteran    FALSE      0.0         Male              17 
4   Amateur     TRUE      0.7         Female            21 
5   Regular     TRUE      0.1         Male              21 
6   Amateur     TRUE      0.0         Female            17 
7   Regular     TRUE      0.0         Female            19 
8   Amateur    FALSE      0.0         Male              21 
9   Amateur     TRUE      0.1         Male              47 
10  Veteran     TRUE      0.0         Female            22 
11  Veteran     TRUE      1.6         Female            23 
12  Pro         TRUE      0.0         Male              17 
13  Beginner    TRUE      1.5         Female            25 
14  Amateur     TRUE      0.2         Male              22 
15  Veteran     TRUE      0.0         Non-binary        17 
16  Amateur    FALSE      0.0         Male              22 
17  Beginner    TRUE      0.0         Female            17 
18  Amateur     TRUE     48.4         Female            17 
19  Amateur     TRUE      0.5         Male              17 
20  Regular     TRUE      0.6         Male              19 
21  Regular     TRUE      0.3         Male              58 
22  Amateur    FALSE      0.1         Female            17 
23  Beginner    TRUE      1.0         Male              17 
24  Beginner    TRUE      0.0         Male              17 
25  Amateur     TRUE      0.7         Female            21 
26  Regular     TRUE      0.6         Female            28 
27  Regular     TRUE      0.0         Male              17 
28  Veteran    FALSE      0.0         Male              23 
29  Amateur     TRUE      1.8         Male              23 
30  Veteran    FALSE      0.1         Male              18 
⋮   ⋮          ⋮         ⋮            ⋮                 ⋮  
167 Amateur    FALSE      1.0         Male              26 
168 Beginner   FALSE      0.3         Female            17 
169 Veteran     TRUE      0.1         Male              17 
170 Veteran    FALSE      0.1         Male              17 
171 Pro        FALSE      0.4         Male              21 
172 Beginner   FALSE      1.8         Male              32 
173 Veteran     TRUE      0.0         Agender           20 
174 Regular    FALSE      0.0         Male              50 
175 Veteran     TRUE      0.0         Male              17 
176 Amateur     TRUE      0.3         Male              17 
177 Pro        FALSE      0.0         Non-binary        17 
178 Veteran    FALSE      2.7         Non-binary        21 
179 Amateur     TRUE      0.4         Two-Spirited      17 
180 Veteran     TRUE      0.1         Non-binary        17 
181 Amateur    FALSE      2.1         Male              24 
182 Amateur     TRUE      0.8         Female            22 
183 Pro        FALSE      0.2         Male              17 
184 Amateur     TRUE     32.0         Male              22 
185 Pro        FALSE      1.7         Male              17 
186 Regular    FALSE      0.1         Male              18 
187 Veteran     TRUE      0.1         Female            44 
188 Amateur     TRUE      0.0         Male              17 
189 Beginner    TRUE      0.0         Female            17 
190 Amateur    FALSE      0.0         Male              17 
191 Amateur     TRUE      0.0         Male              20 
192 Amateur     TRUE      0.0         Female            17 
193 Veteran    FALSE      0.3         Male              22 
194 Amateur    FALSE      0.0         Prefer not to say 57 
195 Amateur    FALSE      2.3         Male              17 
196 Pro         TRUE      0.2         Other             NA

## 1) Numbers of observations/variables in the players.csv dataset

In [32]:
#number of observations/variables (given from source, unaltered):
n_players.csv <- nrow(players_original)
p_players.csv <- ncol(players_original)
cat("players.csv:", n_players.csv, "rows,", p_players.csv, "columns\n")

players.csv: 196 rows, 7 columns


Therefore, in players.csv, there are: 
- 196 observations
- 7 variables

## 2) Variable names, types, and descriptions

In [30]:
desc_of_vars <- c(
    "How the player identifies in gaming experience level",
    "Whether player has subscribed to the newsletter (True or false)",
    "Email address of player",
    "Total time spent in the game (in hours)",
    "Name of player",
    "Gender of player",
    "Age of player (in years)")
   
description_players <- tibble(
    variable = names(players_original),
    type = map_chr(players_original, ~class(.x)[1]),
    description = desc_of_vars)

description_players

variable     type     
experience   experience   character
subscribe    subscribe    logical  
hashedEmail  hashedEmail  character
played_hours played_hours numeric  
name         name         character
gender       gender       character
Age          Age          numeric  
             description                                                    
experience   How the player identifies in gaming experience level           
subscribe    Whether player has subscribed to the newsletter (True or false)
hashedEmail  Email address of player                                        
played_hours Total time spent in the game (in hours)                        
name         Name of player                                                 
gender       Gender of player                                               
Age          Age of player (in years)

#### potential issues
- played_hours: may have extreme outliers
- Age: may have missing values?
- hashedEmail, name: no serious issue but irrelevant to data analysis
- may have inconsistent or duplicated entries

#### possible hidden issues
- "experience" variable is highly biased depending on the person
- sampling bias: only includes players around a certain area
- playtime records may not be accurate due to issues with technology, for example, if playtime was recorded using built-in game timers
- is playtime defined by active play, or even idle/afk play?

#### how the data were collected
- The dataset doesn't specify specificaclly how the data was collected. However, one can assume that the data was taken from activity reports/logs in-game, automatically recorded by the system

## 3) Summary Statistics

In [31]:
summary(players_original)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

Definitions of each term (from the above summary of the dataset):
- length = (in this context) how many rows are in the column
- class = What type of R object it is (category, e.g. numeric, factor, character)
- Mode = how the data is stored in memory (e.g. numeric, character, logical, list)
- Mean = average value
- Median = middle value
- Min/Max = Smallest/largest values
- 1st Qu. = first quarter of the data (first 25% of the dataset), and so on for 2nd, 3rd, 4th Qu.

## 3.1) Means of each numeric variable

In [36]:
numeric_players_data_only <- players_original |>
    select(played_hours, Age)

mean_table <- numeric_players_data_only |>
    summarise(across(everything(), ~ round(mean(.x, na.rm = TRUE), 2))) |>
    pivot_longer(everything(), names_to = "variable", values_to = "mean")

mean_table

variable     mean 
1 played_hours  5.85
2 Age          21.14

The average hours played is 5.85 hours, and the average age is 21.14 years.